From 52dd46448352648db9c2b51fee96b07597aad403 Mon Sep 17 00:00:00 2001
From: Ranjit Waghmode <ranjit.waghmode@xilinx.com>
Date: Fri, 3 Jul 2015 12:15:20 +0530
Subject: [PATCH 251/827] spi: zynqmp: gqspi: remove hardware settings from setup()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

- This patch makes sure that there is no hardware settings
  during the transfer via setup()

- zynqmp_qspi_setup_transfer() call is moved inside function
  zynqmp_qspi_start_transfer()

Signed-off-by: Ranjit Waghmode <ranjit.waghmode@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 0e8578ac513ae4218db9e5457bb4c31470725ee4)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/spi/spi-zynqmp-gqspi.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-zynqmp-gqspi.c b/drivers/spi/spi-zynqmp-gqspi.c
index 0779ac6..564a4bc 100644
--- a/drivers/spi/spi-zynqmp-gqspi.c
+++ b/drivers/spi/spi-zynqmp-gqspi.c
@@ -523,13 +523,13 @@ static int zynqmp_qspi_setup_transfer(struct spi_device *qspi,
  * Sets the operational mode of QSPI controller for the next QSPI transfer,
  * baud rate and divisor value to setup the requested qspi clock.
  *
- * Return:	0 Always
+ * Return:	0 on success; error value otherwise.
  */
 static int zynqmp_qspi_setup(struct spi_device *qspi)
 {
 	if (qspi->master->busy)
 		return -EBUSY;
-	return zynqmp_qspi_setup_transfer(qspi, NULL);
+	return 0;
 }
 
 /**
@@ -842,6 +842,8 @@ static int zynqmp_qspi_start_transfer(struct spi_master *master,
 	xqspi->txbuf = transfer->tx_buf;
 	xqspi->rxbuf = transfer->rx_buf;
 
+	zynqmp_qspi_setup_transfer(qspi, transfer);
+
 	genfifoentry |= xqspi->genfifocs;
 	genfifoentry |= xqspi->genfifobus;
 
-- 
2.9.3

